lib/gen_socket_client.ex

```diff
@@ -316,11 +318,18 @@ defmodule Phoenix.Channels.GenSocketClient do
   defp maybe_connect(:noconnect, state), do: state
 
   defp connect(%{transport_pid: nil} = state) do
-    {:ok, transport_pid} = state.transport_mod.start_link(state.url, state.transport_opts)
+    unless is_nil(URI.parse(state.url).query) do
```

Comment:
extract to named intention declaring helper function? `valid_url?` or something?